diff -ur libcap-ng-0.6.5.orig/utils/filecap.c libcap-ng-0.6.5/utils/filecap.c
--- libcap-ng-0.6.5.orig/utils/filecap.c	2010-09-09 08:43:50.000000000 -0400
+++ libcap-ng-0.6.5/utils/filecap.c	2010-09-09 08:55:25.000000000 -0400
@@ -88,6 +88,7 @@
 	printf("File based capabilities are not supported\n");
 #else
 	char *path_env, *path = NULL, *dir = NULL;
+	struct stat sbuf;
 	int nftw_flags = FTW_PHYS;
 	int i;
 
@@ -104,8 +105,7 @@
 				}
 				return 0;
 			} else if (argv[i][0] == '/') {
-				struct stat buf;
-				if (lstat(argv[i], &buf) != 0) {
+				if (lstat(argv[i], &sbuf) != 0) {
 					printf("Error checking path %s (%s)\n",
 						argv[i], strerror(errno));
 					exit(1);
@@ -113,11 +113,11 @@
 				// Clear all capabilities in case cap strings
 				// follow. If we get a second file we err out
 				// so this is safe
-				if (S_ISREG(buf.st_mode) && path == NULL &&
+				if (S_ISREG(sbuf.st_mode) && path == NULL &&
 								 dir == NULL) {
 					path = argv[i];
 					capng_clear(CAPNG_SELECT_BOTH);
-				} else if (S_ISDIR(buf.st_mode) && path == NULL 
+				} else if (S_ISDIR(sbuf.st_mode) && path == NULL 
 								&& dir == NULL)
 					dir = argv[i];
 				else {
@@ -163,7 +163,7 @@
 		nftw(dir, check_file, 1024, nftw_flags);
 	}else if (path && capabilities == 0) {
 		// Print out specific file
-		check_file(path, NULL, 0, NULL);
+		check_file(path, &sbuf, 0, NULL);
 	} else if (path && capabilities == 1) {
 		// Write capabilities to file
 		int fd = open(path, O_WRONLY|O_NOFOLLOW);
